src/XMakeBuildEngine/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -317,7 +317,7 @@ private string GetProcessesToIgnoreKey(long hostHandshake, long clientHandshake,
         private void ValidateRemotePipeSecurityOnWindows(NamedPipeClientStream nodeStream)
         {
             SecurityIdentifier identifier = WindowsIdentity.GetCurrent().Owner;
-#if FEATURE_PIPE_SECURITY
+#if !FEATURE_PIPE_SECURITY
```

Comment:
Man that's confusing. There's `PipeSecurity` on both sides! This seems reasonable based on your commit notes though. @dsplaisted objections?